### PR TITLE
Doctor P. Holden, Surgeon Miniboss added to the Research Facility Vault

### PR DESCRIPTION
**Pull request by Arthurdentist**
This PR does a few different things. It adds a new civilian mob type, the surgeon, similar to the other hostile civilian mobs we already have in the code.

In addition this PR adds a 'unqiue' surgeon. Hidden deep in space is an abandoned central command facility, where strange and criminal experiments were conducted. The crew of that facility are now dead as a result of these experiments, but one last crew member remains alive, working on his research.

If you find him, you can try and kill him and take his work for yourself, but he wont make it easy for you at all...

surgeon

<h2>This PR isn't finished</h2>

Currently the mob is working, which is why I am uploading what I've done so far now, but the following things need to be done as well.

1. Unique loot needs to be added for him to drop, I have some ideas for a couple of things already, but I would appreciate any other ideas people have.

2. He needs to be placed into the vault

3. The boss surgeon needs to be blacklisted from spawning outside of his vault, (Gold slime cores and other random mob generators).

4. Testing for bugs

Hopefully none of this should take too long to complete.  

<h4> All the beautiful sprites are courtesy of @Blithering </h4>

### CODE REVIEW

**Comment by MadmanMartian** [comment]
We've got a boss mob list for blacklisting big uns if you want to add him to that

_DEFINES/global.dm, line 380

File: code/modules/mob/living/simple_animal/hostile/human/surgeon.dm [hunk 1]
```diff
@@ -1,72 +1,74 @@
-//mob/living/simple_animal/hostile/humanoid/surgeon/
-//	name = "\improper Surgeon"
-//	desc = "Don't interfere with his work!"
-//	icon = 'icons/mob/hostile_humanoid.dmi'
-//	icon_state = "surgeon"
-//	icon_living = "surgeon"
-//	maxHealth = 100
-//	health = 100
+mob/living/simple_animal/hostile/humanoid/surgeon/
```
Comment (jknpj) [inline code/modules/mob/living/simple_animal/hostile/human/surgeon.dm]: /mob/living/simple_animal/hostile/humanoid/surgeon

**Review by jknpj — DISMISSED** [review]

File: code/modules/mob/living/simple_animal/hostile/human/surgeon.dm [hunk 2]
```diff
@@ -1,72 +1,74 @@
...
+	corpse = /obj/effect/landmark/corpse/surgeon
+	items_to_drop = list(/obj/item/weapon/circular_saw/plasmasaw)
 
-//mob/living/simple_animal/hostile/humanoid/surgeon/boss
-//	name = "Doctor Placeholder" //I can't think of a good name
-//	desc = "A deranged surgeon lost deep in space."
-//	icon_dead = null //this isn't even his final form
-//	maxHealth = 80
-//	health = 80
+mob/living/simple_animal/hostile/humanoid/surgeon/boss
```
Comment (jknpj) [inline code/modules/mob/living/simple_animal/hostile/human/surgeon.dm]: missing a / here too

File: code/modules/mob/living/simple_animal/hostile/human/surgeon.dm [hunk 3]
```diff
@@ -1,72 +1,74 @@
...
-//mob/living/simple_animal/hostile/humanoid/surgeon/boss/death(var/gibbed = FALSE)
-//	..(gibbed)
-//	visible_message("<span class=danger><B>Before he can die, the mad surgeon takes a drink of a foul-smelling concoction and begins to mutate! </span></B>")
-//	say("[pick("YOU CAN'T KILL ME THAT EASILY!", "I WONT LET YOU STOP ME!", "GET OUT OF MY FACILITY!", "I MUST CONTINUE MY RESEARCH!", "I'M GONNA WRECK IT!", "I'VE GOT A BONE TO PICK WITH YOU!")]")
-//	var/obj/effect/effect/smoke/S = new /obj/effect/effect/smoke(get_turf(src))
-//	S.time_to_live = 20
-//	new /mob/living/simple_animal/hostile/humanoid/surgeon/skeleton/(get_turf(src))
-//	..()
-//	return qdel(src)
+mob/living/simple_animal/hostile/humanoid/surgeon/boss/death(var/gibbed = FALSE)
```
Comment (jknpj) [inline code/modules/mob/living/simple_animal/hostile/human/surgeon.dm]: ditto

File: code/modules/mob/living/simple_animal/hostile/human/surgeon.dm [hunk 4]
```diff
@@ -1,72 +1,74 @@
...
-//	health = 400
 
-//	move_to_delay = 5 //slow
-//	harm_intent_damage = 30
-//	melee_damage_lower = 25
-//	melee_damage_upper = 35
-//	environment_smash_flags = SMASH_LIGHT_STRUCTURES | SMASH_CONTAINERS | SMASH_WALLS //he will smash through most barriers, but the walls of his vault are rwalls
-//	attacktext = "crushes"
-//	attack_sound = 'sound/weapons/heavysmash.ogg'
+mob/living/simple_animal/hostile/humanoid/surgeon/skeleton
```
Comment (jknpj) [inline code/modules/mob/living/simple_animal/hostile/human/surgeon.dm]: ditto

File: code/modules/mob/living/simple_animal/hostile/human/surgeon.dm [hunk 5]
```diff
@@ -1,72 +1,74 @@
...
-//	faction = "necro"
-//	mob_property_flags = MOB_UNDEAD
-//	items_to_drop = list()
+	corpse = /obj/effect/decal/remains/skelesurgeon
+
+	faction = "necro"
+	mob_property_flags = MOB_UNDEAD
+	items_to_drop = list()
+
+mob/living/simple_animal/hostile/humanoid/surgeon/skeleton/New()
```
Comment (jknpj) [inline code/modules/mob/living/simple_animal/hostile/human/surgeon.dm]: ditto

**Comment by Durgen** [comment]
we don't deserve sprites that nice

**Comment by Arthurdentist** [comment]
pushed some changes that fix the issue pointed out by @jknpj and also added both stages of the boss to the boss list so they'll hopefully be excluded from in-round mob spawners.

File: __DEFINES/global.dm [hunk 6]
```diff
@@ -383,6 +383,8 @@ var/list/boss_mobs = list(
 	/mob/living/simple_animal/hostile/wendigo,				// Stupid strong evolving creature things that scream for help
 	/mob/living/simple_animal/hostile/mechahitler,			// Sieg heil!
 	/mob/living/simple_animal/hostile/alien/queen/large,	// The bigger and beefier version of queens.
+	/mob/living/simple_animal/hostile/humanoid/surgeon/boss, //first stage of Doctor Placeholder
```
Comment (jknpj) [inline __DEFINES/global.dm]: /mob/living/simple_animal/hostile/humanoid/surgeon will take care of both versions

**Comment by Arthurdentist** [comment]
alright, big boss now drops a loot crate when he dies, which (so far) contains a morph mask to turn into a giant skellington and a bunch of medication to help you not die from the fight.

Im not sure if I have more time to do loot at the moment, but this first bit is done at any rate.

**Comment by Blithering** [comment]
One loot item that would fit the theme and wouldn't require more coding would be an unlabelled wizard potion.

**Comment by Arthurdentist** [comment]
alright, going to say this is as finished as its going to get.

The boss now drops a 'loot crate' when he dies. Inside you will find some drops, you currently get a morph mask, a guaranteed healing potion and one other, random magical potion. I've left it so that more random loot can be added in the future, but im running out of time to code additional things for this PR and I'd like to come back to it at another date. I've also added him to his lair so we should be good to go now.

**Pull request merged**